File: benchmarks/values/bigint_fib.py
import time
import sys
sys.set_int_max_str_digits(100000)

t0 = time.perf_counter()

a = 0
b = 1
for i in range(100000):
    a, b = b, a + b

digits = len(str(b))

t1 = time.perf_counter()
print(digits)
print(f"elapsed: {t1 - t0:.3f}s")

File: benchmarks/suite/bigint_fib.go
package main

import (
	"fmt"
	"math/big"
	"time"
)

func main() {
	t0 := time.Now()

	a := big.NewInt(0)
	b := big.NewInt(1)
	tmp := new(big.Int)

	for i := 0; i < 100000; i++ {
		tmp.Set(b)
		b.Add(a, b)
		a.Set(tmp)
	}

	digits := len(b.String())

	elapsed := time.Since(t0)
	fmt.Println(digits)
	fmt.Printf("elapsed: %.3fs\n", elapsed.Seconds())
}
